src/column/reader.rs

```diff
@@ -671,8 +701,11 @@ mod tests {
             num_values_cur_page += 1;
           }
         }
-      } else {
+      } else if levels_per_page > 0 {
         num_values_cur_page = levels_per_page;
+      } else {
+        // for testing required non-repeated columns (levels_per_page == 0)
+        num_values_cur_page = batch_size;
```

Comment:
It looks like `num_values_cur_page` was only set when levels per page are defined. In my case I set it explicitly to 0, therefore I have to update number of values in page to propagated `batch_size`. Not sure, if it is a good way of doing it.

Reply:
It seems your tests still use `NUM_LEVELS` instead of 0? Also `batch_size` is not added to the `make_pages` function so maybe the build won't pass.

Also I think there's another stupid bug of mine in the following that may be related:
```rust
let mut page_builder = DataPageBuilderImpl::new(desc.clone());
if max_def_level > 0 {
  page_builder.add_rep_levels(max_rep_level, &rep_levels[level_range.clone()]);
}
if max_rep_level > 0 {
  page_builder.add_def_levels(max_def_level, &def_levels[level_range]);
}
```
It should process def levels in the first if and rep levels in the second if.

Reply:
Yes, you are right. I think it is still uses NUM_LEVELS to get values per page, not batch size. I will have a closer look at `make_pages` function and will also update code with your changes, and will try to come up with unittest for it. 

I am super confused with definition and repetition levels at this point:) Need to spend some time figuring out which is which.